eventtracking/backends/routing.py

```diff
@@ -0,0 +1,132 @@
+"""Route events to processors and backends"""
+
+from collections import OrderedDict
+import logging
+
+from eventtracking.processors.exceptions import EventEmissionExit
+
+LOG = logging.getLogger(__name__)
+
+
+class RoutingBackend(object):
+    """
+
+    Route events to the appropriate backends.
+
+    A routing backend has two types of components:
+
+    1) Processors - These are run sequentially, processing the output of the previous processor. If you had three
+       processors [a, b, c], the output of the processing step would be `c(b(a(event)))`. Note that for performance
+       reasons, the processor is able to actually mutate the event dictionary in-place. Event dictionaries may be large
+       and highly nested, so creating multiple copies could be problematic. A processor can also choose to prevent the
+       event from being emitted by raising `EventEmissionExit`. Doing so will prevent any subsequent processors from
+       running and prevent the event from being sent to the backends. Any other exception raised by a processor will be
+       logged and swallowed, subsequent processors will execute and the event will be emitted.
+    2) Backends - Backends are intended to not mutate the event and each receive the same event data. They are not
+       chained like processors. Once an event has been processed by the processor chain, it is passed to each backend in
+       the order that they were registered. Backends typically persist the event in some way, either by sending it
+       to an external system or saving it to disk. They are called synchronously and in sequence, so a long running
+       backend will block other backends until it is done persisting the event. Note that you can register another
+       `RoutingBackend` as a backend of a `RoutingBackend`, allowing for arbitrary processing trees.
+
+    `backends` is a collection that supports iteration over its items using `iteritems()`. The keys are expected to be
+        sortable and the values are expected to expose a `send(event)` method that will be called for each event. Each
+        backend in this collection is registered in order sorted alphanumeric ascending by key.
+    `processors` is an iterable of callables.
+
+    Raises a `ValueError` if any of the provided backends do not have a callable "send" attribute or any of the
+        processors are not callable.
+    """
+
+    def __init__(self, backends=None, processors=None):
+        self.backends = OrderedDict()
+        self.processors = []
+
+        if backends is not None:
+            for name in sorted(backends.keys()):
```

Comment:
I don't understand why this is necessary, if backends is an OrderedDict?   Why does it matter in what order they are registered?